flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsCheckpointMetadataOutputStream.java

```diff
@@ -108,8 +107,8 @@ public void close() {
             closed = true;
 
             try {
-                out.close();
-                fileSystem.delete(metadataFilePath, false);
+                outputStreamWrapper.close();
+                outputStreamWrapper.abort();
```

Comment:
`cleanup` might be better in consider of the calling sequences here?
Also cc @Myasuka

Reply:
Agreed, thanks for the suggestion.